php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -576,6 +576,15 @@ public function clear()
         }
     }
 
+    /**
+     * Clear all unknown fields previously parsed.
+     * @return null.
+     */
+    public function discardUnknownFields()
+    {
+        $this->unknown = "";
```

Comment:
it should recursively clear all the unknown fields in the sub messages.

Reply:
What about map fields?

Reply:
does map entry itself preserve unknown?